### PR TITLE
[BUGFIX] Core.addBlock() to accept transactions with older mixin values for a period of time.

### PR DESCRIPTION
**Pull request by brandonlehmann**
Hopefully addresses #491

### CODE REVIEW

File: src/CryptoNoteCore/Core.cpp [hunk 1]
```diff
@@ -588,7 +588,11 @@ std::error_code Core::addBlock(const CachedBlock& cachedBlock, RawBlock&& rawBlo
     return error::BlockValidationError::DIFFICULTY_OVERHEAD;
   }
 
-  if (!validateMixin(transactions, blockIndex))
+  // This allows us to accept blocks with transaction mixins for the mined money unlock window
+  // that may be using older mixin rules on the network. This helps to clear out the transaction
+  // pool during a network soft fork that requires a mixin lower or upper bound change
+  uint32_t mixinChangeWindow = blockIndex - CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW;
```
Comment (zpalmtree) [inline src/CryptoNoteCore/Core.cpp]: This could be negative if we're at the start of a chain.

Comment (brandonlehmann) [reply src/CryptoNoteCore/Core.cpp]: *sigh*

Comment (brandonlehmann) [reply src/CryptoNoteCore/Core.cpp]: That work?

Comment (zpalmtree) [reply src/CryptoNoteCore/Core.cpp]: Yep

**Comment by brandonlehmann** [comment]
Forgot a `;`

**Pull request merged**